docs: add code review guide (#2772)
kanisterio/kanister: CODE_REVIEW.md

# Code review requiements

This document describe responsibilities of code reviewers when reviewing PRs.

Pull request process is described in [contributing guide](./CONTRIBUTING.md#submitting-pull-requests)

## Base checklist

- All automated test steps pass (e.g. tests, lints, build)
- PR title follows [commit conventions](CONTRIBUTING.md#commit-conventions)
    - If PR format is different, reviewer should change it to follow the conventions
- PR has a description with reasoning and change overview
    - If description is missing but clear for reviewer, reviewer may add a comment or edit description
- New feature or fix has tests proving it works
    - Reviewer should request changes from contributor to add tests
- If PR changes documented behaviour, there should be update to `/docs` or inline comments

>     - Reviewer should request changes from contributor to add tests

Reviewer: ```suggestion
    - Reviewer should request changes from contributor to update docs
```
I am guessing this is what you meant to add here?

- If PR introduces breaking changes, fixes a bug or adds a new feature, there should be a [release note](#release-notes)

>     - Reviewer may add a release note by themself

Reviewer 2: I think we should clarify, whose responsibility it is to add the release note. I think it's PR author's responsibility. if we  not explicitly set the expectation here, one would think the other is going to do that.

Author: Good point. Added to contributors guide and mentioned here.

## Requesting changes

It's recommended to request changes by submitting `comment` type reviews.
`Request changes` type review would block merging until requester approves the
changes, this can slow down the process if there are multiple reviewers.

## Approving and merging

We use `kueue` bot to merge approved PRs.

If PR is approved, all checks are passing and it has the `kueue` label, it will

> be automatically suqashed and merged.

Reviewer: ```suggestion
be automatically squashed and merged.
```

> For PRs from kanister developers, the committer should add the `kueue` label after

Reviewer: ```suggestion
For PRs from Kanister developers, the committer should add the `kueue` label after
```

PR was approved.

For PRs from community, the reviewer should add the `kueue` label.

## Release notes

> Kanister is using [reno](https://docs.openstack.org/reno/latest/) tool to generate changelogs from release note files.

Reviewer: ```suggestion
Kanister is using the [reno](https://docs.openstack.org/reno/latest/) tool to generate changelogs from release note files.
```
